src/pages/organize/[orgId]/people/duplicates/index.tsx

```diff
@@ -45,20 +59,39 @@ const DuplicatesPage: PageWithLayout = () => {
           </Typography>
         </Box>
       )}
-      {list.data && list.data.length > 0 && (
-        <Box p={1.5}>
+      {totalItems > 0 && (
+        <Box
+          ref={totalPages > 1 ? containerRef : undefined}
+          sx={{
+            p: 1.5,
+            ...(totalPages > 1 && {
+              maxHeight: '80vh',
+              overflowY: 'auto',
+            }),
+          }}
+        >
           <Typography
             color={oldTheme.palette.grey[500]}
             sx={{ mb: 2, textTransform: 'uppercase' }}
             variant="subtitle2"
           >
             {messages.page.possibleDuplicates()}
           </Typography>
-          {list.data
-            .filter((cluster) => cluster.status === 'pending')
-            .map((cluster) => (
-              <DuplicateCard key={cluster.id} cluster={cluster} />
-            ))}
+
+          {paginatedData.map((cluster) => (
+            <DuplicateCard key={cluster.id} cluster={cluster} />
+          ))}
+
+          <Box display="flex" justifyContent="center" mt={3}>
+            <Pagination
+              count={totalPages}
+              onChange={(_, value) => {
+                setPage(value);
+                containerRef.current?.scrollTo({ behavior: 'smooth', top: 0 });
```

Comment:
This does not do anything unless the container is scrolled, which it won't be with the change I'm proposing above. Consider using `scrollIntoView()` instead.